faker/providers/internet/__init__.py

```diff
@@ -169,20 +199,50 @@ def url(self):
         pattern = self.random_element(self.url_formats)
         return self.generator.parse(pattern)
 
-    def ipv4(self, network=False):
-        """Produce a random IPv4 address or network with a valid CIDR"""
-        address = str(ip_address(self.generator.random.randint(
-            0, (2 ** IPV4LENGTH) - 1)))
-        if network:
+    def ipv4(self, network=False, private=False):
+        """Produce a random IPv4 address or network with a valid CIDR."""
+        if private is True:
+            networks = dict()
+            for i in range(len(_private_ipv4_networks)):
+                networks[i] = _private_ipv4_networks[i].num_addresses
+            k = self.generator.random_element(networks)
+            _net = _private_ipv4_networks[k]
+            address = str(ip_address(self.generator.random.randint(
+                int(_net.network_address),
+                int(_net.network_address) + _net.num_addresses - 1
+            )))
+        else:
+            # It's unfair to trying to not find private ip but it's much faster
+            address = str(ip_address(self.generator.random.randint(
+                0, (2 ** IPV4LENGTH) - 1)))
+            while ip_address(force_text(address)).is_private is True:
```

Comment:
I would default the `private` argument to `None`, and have this check only `if private is False`